CMakeLists.txt

```diff
@@ -18,6 +18,8 @@ option(S2N_NO_PQ_ASM "Turns off the ASM for PQ Crypto even if it's available for
 You likely want this on older compilers." OFF)
 option(SEARCH_LIBCRYPTO "Set this if you want to let S2N search libcrypto for you,
 otherwise a crypto target needs to be defined." ON)
+option(UNSAFE_IGNORE_COMPILER_WARNINGS "Compiler warning is treated as an error. Try turning
+this off when observing errors on a new or unpopular compiler." ON)
```

Comment:
Can we include a warning that ignoring warnings may lead to ignoring unexpected/untested behavior?